pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py

```diff
@@ -181,7 +181,7 @@ def cache_training_step_metrics(self, opt_closure_result):
         self.logged_metrics.update(logged_metrics_tmp)
         self.cached_results.legacy_batch_log_metrics.update(logged_metrics_tmp)
 
-    def log_metrics(self, metrics, grad_norm_dic, step=None):
+    def log_metrics(self, metrics, grad_norm_dic, step=None, log_train_step_metrics=False):
```

Comment:
lets add it in docs and explain how it fuse with `step` argument
- what happen in `step=n` and `log_train_step_metrics=False`
- what happen in `step=None` and `log_train_step_metrics=True`

Reply:
shall it be `log_train_metrics_kind='step'|'epoch'`?

Reply:
Hey @jirka, 
`log_train_step_metrics` is there only for internal reason. 
This should not be exposed in the doc or made over-complicated :)
I wanted to have a simple boolean to keep things fast.